lib/rules/object-shorthand.js

```diff
@@ -264,10 +264,13 @@ module.exports = {
                     keyPrefix + keyText + sourceCode.text.slice(tokenBeforeParams.range[1], node.value.range[1])
                 );
             }
+
             const arrowToken = sourceCode.getTokenBefore(node.value.body, { filter: token => token.value === "=>" });
             const tokenBeforeArrow = sourceCode.getTokenBefore(arrowToken);
-            const hasParensAroundParameters = tokenBeforeArrow.type === "Punctuator" && tokenBeforeArrow.value === ")";
-            const oldParamText = sourceCode.text.slice(sourceCode.getFirstToken(node.value, node.value.async ? 1 : 0).range[0], tokenBeforeArrow.range[1]);
+
+            const firstToken = sourceCode.getFirstToken(node.value, node.value.async ? 1 : 0);
```

Comment:
I'm a little concerned that this could be brittle if more arrow function modifiers are added in future language versions.

I'm wondering if it would be possible to search for tokens between the first token of `node.value` (inclusive), and the first token of `node.value.params[0]` (exclusive)? `sourceCode.getTokensBetween()` does an exclusive search, so maybe it could be something like `sourceCode.getTokensBetween(node.key, node.value.params[0], { filter: astUtils.isOpenParenToken })`?

If it's too much of a pain, I'm okay with what you have now, but I wanted to raise the question.